mesonbuild/build.py

```diff
@@ -1818,25 +1816,27 @@ def post_init(self) -> None:
         if not hasattr(self, 'suffix'):
             # Executable for Windows or C#/Mono
             if machine.is_windows() or machine.is_cygwin() or 'cs' in self.compilers:
-                self.suffix = 'exe'
+                self.suffix = '.exe'
```

Comment:
I think it would be best not to change this. Instead where the suffix is concatenated to the file name, conditionally add a dot.